### DISCUSSION TITLE
pulseIn() seems to reset my ATTINY84 and 85

### DISCUSSION THREAD

**Discussion by pone7777** [SpenceKonde/ATTinyCore#839] [Support / Q & A]
When I use pulseIn() my ATTiny84 resets and I cant seem to figure out why?  I'm testing with this simple code on an ATTiny84 @ 8MHz internal OSC, millis/micros enabled.

```
int pin = PIN_PA5;
unsigned long duration;

void setup() {
  Serial.begin(9600);
  pinMode(pin, INPUT);
  Serial.println("setup... ");
}

void loop() {
  duration = pulseIn(pin, HIGH, 100000);
  Serial.println(duration);
  delay(500);
}
```

and all I seem to get is :

```
setup... 
setup... 
setup... 
setup... 
```
suggesting to me that the MCU is resetting?  I get the same behaviour with an ATTiny85.
This code works as expected on a Nano, resulting in:

```
setup... 
10300
10277
10449
10310
```
Am I missing something obvious? Thanks in advance for your help!

**Comment by hmeijdam** [comment]
Just tested on a ATTiny85 (PIN_PB3) and works perfectly fine. I hooked up a servo tester to confirm it's pulses are measured accurately. 
Your problem must be elsewhere.

**Reply by pone7777** [reply]
Thanks for testing. I'm stumped. I've got the decoupling capacitors, the Vcc line has less than 40mV ripple and ive tried 5V and 3.3V, the fuses are pretty standard: L0xE2, H0xDF, E0xFE.

I'm out of ideas!

**Reply by hmeijdam** [reply]
I have compiled your sketch with these settings and the hex file is below.

I had to rename it from "pulsein_test.ino.hex" to pulsein_test.ono.txt to be able to upload it here. So you need to rename it back to the "hex" extension. Then you can load it into your AT84 to test. I use [Avrdudess](https://github.com/ZakKemble/AVRDUDESS/releases) to load hexfiles 

![image](https://github.com/SpenceKonde/ATTinyCore/assets/9595465/c7995fa5-4195-4314-a16b-35cf3a38a02d)

[Pulsein_test.ino.txt](https://github.com/SpenceKonde/ATTinyCore/files/14250947/Pulsein_test.ino.txt)

**Reply by pone7777** [reply]
Hans,
THANK YOU!!!  Your .hex worked, so I knew it wasnt an issue with my particular ATTinys.  The PNG image of the settings that you used is what actaully solved my problem: I noticed I didnt have an option for ATTiny84a (only listed TinyCore ATTiny84) nor LTO, so I assumed I've somehow installed an outdated(?) version of TinyCore.  Just did a clean install, and my own sketched using pulseIn() now work!

Many many thanks for taking the time to help out, it's very much appreciated!  Now I can carry on with my projects.

-Tony

**Reply by hmeijdam** [reply]
Good to hear it's solved. Good luck with your project.
